gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -1061,6 +1062,33 @@ private void handleListenerRegistration(final HttpServerRequest request) {
         });
     }
 
+    private boolean isListenerJsonInvalid(HttpServerRequest request, Buffer hookData) {
+        if( isHookJsonInvalid(request, hookData)){
+            // No further checks required. hook definitively is invalid.
+            return true;
+        }
+
+        final JsonObject hook;
+        try {
+            // Badly we need to parse that JSON one more time.
+            hook = new JsonObject(hookData);
+        } catch (DecodeException e) {
+            log.error("Cannot decode JSON", e);
+            badRequest(request, "Cannot decode JSON", e.getMessage());
+            return true;
+        }
+        final JsonArray methods = hook.getJsonArray("methods");
+        for (Object method : methods) {
+            if( !QueueProcessor.httpMethodIsQueueable( HttpMethod.valueOf((String)method) ) ){
+                final String msg = "Listener registration request tries to hook for forbidden '"+method+"' method.";
+                log.error( msg );
+                badRequest(request, "Bad Request", msg+"\n" );
```

Comment:
Do not respond this requests with "400 Bad Request". Use "405 Method not Allowed"

Reply:
Good point. I did not thought about this. Therefore it's OK to respond with a 400